src/els_code_navigation.erl

```diff
@@ -74,6 +86,55 @@ goto_definition(_Uri, #{ kind := parse_transform, id := Module }) ->
 goto_definition(_Filename, _) ->
   {error, not_found}.
 
+-spec is_imported_bif(uri(), atom(), non_neg_integer()) -> boolean().
+is_imported_bif(Uri, F, A) ->
+  OldBif = erl_internal:old_bif(F, A),
+  Bif = erl_internal:bif(F, A),
+  case {OldBif, Bif} of
+    %% Cannot be shadowed, always imported
+    {true, true} ->
+      true;
+    %% It's not a BIF at all
+    {false, false} ->
+      false;
+    %% The hard case
+    {_, _} ->
+      is_bif_auto_import_disabled(Uri, F, A)
```

Comment:
my personal opinion (but I am not an erlang_ls maintainer) is that you don't need to check if auto_import is disabled, just jump to the bif at this point
- it can be disabled also in a header file, or rebar.config or erlc command line option of which erlang_ls might not even know
- auto_import is usually disabled to silence warnings about a bif clashing with a local function - but at this point no local function is found so it is highly likely there won't be a no_auto_import directive

what I mean is that this algorithm can not be 100% correct (because erlang_ls does not have all the info) so instead let's use a simple heuristic that already covers most of the practical cases (and also not 100% correct)

btw very nice thorough code covering all the edge cases